packages/flutter/lib/src/widgets/primary_scroll_controller.dart

```diff
@@ -34,14 +49,18 @@ class PrimaryScrollController extends InheritedWidget {
   const PrimaryScrollController({
     super.key,
     required ScrollController this.controller,
+    this.autoForPlatforms = _kMobilePlatforms,
+    this.scrollDirection = Axis.vertical,
```

Comment:
> whether the PrimaryScrollController will be inherited

Do you mean the automatic inheritance? This can be easily confused I think in talking about this. I do not like that the PSC is automatically inherited _in some cases_ but it is the behavior we already have baked in. 

Any ScrollView can inherit the PSC if they want to using `primary`, these only relate to defining the case for automatically inheriting.

Reply:
> Does this need to be a collection? What if I want it for both scroll directions?

I don't think we should allow this for automatic inheritance. It would automatically attach to every ScrollView in every direction and I cannot think of a valid case for having that. If the PSC was accessed to animate the position etc, you would run into the same issue we are trying to solve - that more than one position is attached.